Tests/Tests/IncrementalStore.m

```diff
@@ -731,6 +734,144 @@ - (void)testFetchConstraints
     }
 }
 
+- (void)testBatchUpdates
+{
+    int num_entries = 100;
+
+    XCTAssertTrue((num_entries % 4) == 0, @"Test assumes num_entries is mod 4: %d", num_entries);
+
+    NSError *err = nil;
+    // This will create the database
+    NSManagedObjectContext *moc = self.managedObjectContext;
+    XCTAssertNotNil(moc, @"could not create Context");
+
+    for (int i = 0; i < num_entries; i++) {
+        Entry *e = MakeEntry(moc);
+        e.text = NSStringFromSelector(_cmd);
+        e.check = (i % 2) ? @NO : @YES;
+        e.i64 = @(i);
+    }
+
+    // push it out
+    XCTAssertTrue([moc save:&err], @"Save Failed: %@", err);
+
+    NSArray *results;
+    /**
+     *  Fetch checked entries
+     */
+    NSPredicate *checked = [NSPredicate predicateWithFormat:@"check == YES"];
+
+    NSFetchRequest *fr = [NSFetchRequest fetchRequestWithEntityName:@"Entry"];
+    fr.shouldRefreshRefetchedObjects = YES;
+    fr.predicate = checked;
+
+    err = nil;
+    results = [moc executeFetchRequest:fr error:&err];
+    XCTAssertNotNil(results, @"Expected results: %@", err);
+
+    XCTAssertTrue([results count] == (num_entries / 2), @"results count should be %d is %lu",
+                  num_entries / 2, (unsigned long)[results count]);
+
+    for (Entry *e in results) {
+        XCTAssertTrue([e.check boolValue], @"not even?");
+
+        long long val = [e.i64 longLongValue];
+        XCTAssertTrue((val % 2) == 0, @"entry.i64 should be even");
+    }
+
+    /**
+     *  Batch update all objects 50 or higher (should remove 25 checks)
+     */
+    {
+        NSBatchUpdateRequest *req = [[NSBatchUpdateRequest alloc] initWithEntityName:@"Entry"];
+        req.predicate = [NSPredicate predicateWithFormat:@"i64>=%d", num_entries / 2];
+        req.propertiesToUpdate = @{ @"check" : @(NO) };
+        req.resultType = NSUpdatedObjectsCountResultType;
+        NSBatchUpdateResult *res = (NSBatchUpdateResult *)[moc executeRequest:req error:&err];
+
+        if (!CDTISSupportBatchUpdates) {
+            XCTAssertNil(res, @"Result should be nil since batch updates are not supported");
+            XCTAssertTrue([err.domain isEqualToString:CDTISErrorDomain],
+                          @"Error domain should indicate error source");
+            XCTAssertTrue(err.code == CDTISErrorExectueRequestTypeUnkown,
+                          @"Error code should identify error reason");
```

Comment:
I think you should just `return` from this test here.  There is no sense to continue really.